sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -431,6 +455,252 @@ function main() {
 }
 ```
 
+## Advanced Examples
+
+### Custom Credentials
+
+The `@azure/identity` library covers a broad range of Azure Active Directory authentication scenarios. However, it's possible the credential implementations provided might not meet the specific needs your application, or an application might want to avoid taking a dependency on the `@azure/identity` library.
+
+In this section we'll examine some example cases in which it might make sense to write a credential on your own.
+
+### Authenticating with a pre-fetched access token
+
+Our package `@azure/core-http` exports a `TokenCredential` interface which is used by the `@azure/identity` package to define a common public API for all of the Identity credentials that we offer.
+
+The `@azure/identity` library does not contain a `TokenCredential` implementation which can be constructed directly with an `AccessToken`. This is intentionally omitted as a main line scenario as access tokens expire frequently and have constrained usage. However, we understand there may be some scenarios where authenticating a service client with a pre-fetched token is necessary.
+
+In this example, `StaticTokenCredential` implements the `TokenCredential` abstraction. It takes a pre-fetched access token in its constructor as an `AccessToken` (defined on `@azure/core-http`), and simply returns that from its implementation of `getToken()`.
+
+```ts
+import { TokenCredential, AccessToken } from "@azure/core-http";
```

Comment:
We have been very intentional when showing samples to use JavaScript rather than TypeScript. We have the TS samples only in our samples folder where we can afford to keep both TS & JS samples. Can we do the same here?

Reply:
For this PR I’d like to keep this in typescript, as it will mean double the files, and if we want to cater specifically to either audience, it may make sense to give more context on what typescript dependencies to install, for example.

Reply:
Well, let’s do this: I can add the two documents as part of this PR, but let’s get one of them approved first (the typescript one, which will be more complex), then I’ll do the JS one.

Reply:
On second thought, two documents won’t work well with the samples v2 approach. We have an issue to migrate to samples v2. I believe we can have one file for now and improve over time.